src/cpp/desktop/DesktopMainWindow.cpp

```diff
@@ -663,10 +673,15 @@ void MainWindow::onLoadFinished(bool ok)
    std::ostringstream oss;
    Error error = text::renderTemplate(options().resourcesPath().completePath("html/connect.html"),
                                       vars, oss);
-   if (error)
-      LOG_ERROR(error);
-   else
-      loadHtml(QString::fromStdString(oss.str()));
+
+   LOCK_MUTEX(mutex_)
+   {
+      if (error)
+         LOG_ERROR(error);
+      else if (!isErrorDisplayed_)
+         loadHtml(QString::fromStdString(oss.str()));
```

Comment:
Nit: I think the whole template rendering bit should be in here since none of it is meaningful work if the error is already displayed.